test/Emission.test.ts

```diff
@@ -0,0 +1,129 @@
+  });
+
+  before(async function () {
+    const chainId = hre.network.config.chainId;
+    if (!chainId) {
+      throw new Error('Chain ID not found');
+    }
+
+    governanceAddresses = mento.getContractsByChainId(chainId);
+    if (!governanceAddresses) {
+      throw new Error('Governance addresses not found for this chain');
+    }
+
+    emission = Emission__factory.connect(
+      governanceAddresses.Emission,
+      // @TODO: Do I need this line below?
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      provider as any,
+    );
+
+    mentoToken = MentoToken__factory.connect(
+      governanceAddresses.MentoToken,
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      provider as any,
+    );
+
+    console.log('\r\n========================');
+    console.log(
+      'Running Emission Contract tests on network with chain id:',
+      chainId,
+    );
+    console.log('========================\r\n');
+  });
+
+  describe('emission.calculateEmission()', async function () {
+    for (const [period, timeToTravel, expectedEmission] of EMISSION_SCHEDULE) {
+      it(`should calculate the correct amount after ${period}`, async function () {
+        const start = await emission.emissionStartTime();
+        const now = await helpers.time.latest();
+        const elapsed = BigInt(now) - start;
+
+        if (elapsed >= timeToTravel) {
+          // can only time travel to the future
+          this.skip();
+        }
+
+        await helpers.time.increase(BigInt(timeToTravel) - elapsed);
+        expect(BigInt(await helpers.time.latest()) - start).to.eq(timeToTravel);
+
+        expect(await emission.calculateEmission()).to.equal(expectedEmission);
+      });
+    }
+  });
+
+  describe('emission.emitTokens()', async function () {
+    for (const [period, timeToTravel, expectedEmission] of EMISSION_SCHEDULE) {
+      it(`should emit the correct amount after ${period}`, async function () {
+        const start = await emission.emissionStartTime();
+        const now = await helpers.time.latest();
+        const elapsed = BigInt(now) - start;
+
+        if (elapsed >= timeToTravel) {
+          // can only time travel to the future
+          this.skip();
+        }
+
+        // emission.emitTokens() will increase block.timestamp by 1 second,
+        // therefore to mint at the exact time, we need to subtract 1s
+        await helpers.time.increase(BigInt(timeToTravel) - elapsed - BigInt(1));
+        expect(BigInt(await helpers.time.latest()) - start).to.eq(
+          timeToTravel - 1,
+        );
+
+        const emittedSoFar = await emission.totalEmittedAmount();
+        expect(await mentoToken.emittedAmount()).to.equal(emittedSoFar);
+
+        const [signer] = await ethers.getSigners();
+        await emission.connect(signer!).emitTokens();
+        expect(BigInt(await helpers.time.latest()) - start).to.eq(timeToTravel);
+
+        expect(await emission.totalEmittedAmount()).to.equal(
```

Comment:
NIT: lets stick to .eq or .equal

Reply:
I removed the block.timestamp asserts so it should be `.equal` everywhere now